css-positioning/css-positioning_layout-with-position/css/styles.css

```diff
@@ -0,0 +1,108 @@
+*,
+*::before,
+*::after {
+  box-sizing: border-box;
+}
+
+body {
+  display: flex;
+  flex-direction: column;
+  margin: 80px 0 45px 0;
+  font-family: system-ui;
+}
+
+/**
+ *
+ * Place the header at the top of the viewport and make sure, that it remains there when the
+ * document is scrolled.
+ */
+.header {
+  background: #8236ff;
+  color: #ffffff;
+  height: 80px;
+  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
+  padding: 10px;
+  position: fixed;
+  top: 0;
+  z-index: 1;
+  width: 100%;
+  text-align: center;
+}
+
+.header__text {
+  color: #ffdee2;
+}
+
+.header__heading {
+  margin: 0;
+}
+
+/**
+ *
+ * Place the footer at the bottom of the viewport and make sure, that it remains there when the
+ * document is scrolled.
+ */
+.footer {
+  background: #7fd9e2;
+  color: #ffffff;
+  height: 48px;
+  width: 100%;
+  position: fixed;
+  text-align: center;
+  bottom: 0;
+  padding-top: 10px;
+}
+
+/**
+ *
+ * Define the card as reference frame for descendant elements, that require absolute positioning.
+ */
+.card {
+  margin: 20px 10%;
+  border: 5px solid #8236ff;
+  padding: 3% 10%;
+  scroll-margin-top: 100px;
+  border-radius: 2rem;
+  position: static;
+}
+
+.card__heading {
+  margin: 0 0 10px 0;
+}
+
+/**
+ * Place the bookmark in the top, right corner of each card.
+ */
+.card__bookmark {
+  border: 2px solid #8236ff;
+  font-size: 2em;
+  padding: 6px 12px;
+  border-radius: 50%;
+  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
+  position: relative;
+  left: 940px;
+  bottom: 65px;
+}
+
+/**
+ * The go-to-top button should cover all other elements, so that it is always visible, and
+ * unaffected when the document is scrolled.
+ */
+.goto-top {
+  width: 48px;
+  height: 48px;
+  padding: 8px;
+  border-radius: 50%;
+  background: #8236ff;
+  color: white;
+  box-shadow: 0 6px 8px rgba(0, 0, 0, 0.4);
+  text-decoration: none;
+  flex-shrink: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: fixed;
+  z-index: 1;
```

Comment:
And also the button is behind the footer. I am not sure if this is because your header is already with z-index 1. Maybe try something else, but I don't really know why is that.